hollow/src/main/java/com/netflix/hollow/api/client/HollowDataHolder.java

```diff
@@ -177,7 +181,21 @@ private void applyStateEngineTransition(HollowBlobInput in, OptionalBlobPartInpu
             reader.applyDelta(in, optionalPartIn);
         }
 
-        setVersion(transition.getToVersion());
+        long expectedToVersion = transition.getToVersion();
+        Long actualToVersion = null;
+        String actualToVersionStr = stateEngine.getHeaderTag(HEADER_TAG_PRODUCER_TO_VERSION);
+        if (actualToVersionStr != null) {
+            actualToVersion = Long.valueOf(actualToVersionStr);
+        }
+
+        if (actualToVersion != null) {
+            if (actualToVersion.longValue() != expectedToVersion) {
+                LOG.warning("toVersion in blob didn't match toVersion seen in metadata; actualToVersion=" + actualToVersion + ", expectedToVersion=" + expectedToVersion);
```

Comment:
nit: the terms blob and metadata dont apply in this local context as we want them to, we can rephrase the msg to something like
```
toVersion encoded in incoming blob header did not match toVersion requested in transition; the one from blob header will be assumed
```

Reply:
not apply anymore